commands/modmail.py

```diff
@@ -35,10 +35,143 @@ async def cancel(self, button: discord.ui.Button, interaction: discord.Interacti
+        self.guild = guild
+        self.bot = bot
+
+    @discord.ui.button(
+        label="Send a modmail!",
+        style=discord.ButtonStyle.primary,
+        custom_id="persistent_view:sendmodmail",
+    )
+    async def send_modmail(self, button: discord.ui.Button, interaction: discord.Interaction):
+        async def modmail_prompt(listen_channel: discord.DMChannel):
+            def check(m):
+                return m.author == interaction.user and m.channel == listen_channel
+            try:
+                await listen_channel.send(embed=gen_embed(title='Modmail Message Contents',
+                                           content='Please type out your modmail below and send. You can send images by adding an attachement to the message you send.'))
+            except discord.Forbidden:
+                raise RuntimeError('Forbidden 403 - could not send direct message to user.')
+
+            try:
+                mmsg = await self.bot.wait_for('message', check=check, timeout=300.0)
+            except asyncio.TimeoutError:
+                await listen_channel.send(embed=gen_embed(title='Modmail Cancelled',
+                                               content='The modmail has been cancelled.'))
+                return None
+            return mmsg
+
+        await interaction.response.defer()
+        log.info('Begin modmail interaction workflow')
+        dm_channel = interaction.user.dm_channel
+        if not dm_channel:
+            dm_channel = await interaction.user.create_dm()
+        modmail_content = await modmail_prompt(dm_channel)
+
+        if modmail_content:
+            log.info('Message detected from user')
+            view = Confirm()
+            sent_message = await dm_channel.send(embed=gen_embed(title='Are you sure you want to send this?',
+                                                              content='Please verify the contents before confirming.'),
+                                              view=view)
+            timeout = await view.wait()
+            if timeout:
+                log.info('Confirmation view timed out')
+                for item in view.children:
+                    item.disabled = True
+                    await sent_message.edit(embed=gen_embed(title='Are you sure you want to send this?',
+                                                            content='Please verify the contents before confirming.'),
+                                            view=view)
+                    return
+            await sent_message.edit(embed=gen_embed(title='Are you sure you want to send this?',
+                                                              content='Please verify the contents before confirming.'),
+                                              view=view)
+
+            if view.value:
+                log.info('Workflow confirm, compilation and send logic start')
+                document = await db.servers.find_one({"server_id": self.guild.id})
```

Comment:
```suggestion
                document = await db.servers.find_one({"server_id": interaction.guild.id})
```

We can bypass the need to store guild in the view since the interaction is initiated in the guild when they click the button.